devlib/module/cgroups.py

```diff
@@ -168,7 +169,31 @@ def move_all_tasks_to(self, dest, exclude=[]):
             if cgroup != dest:
                 self.move_tasks(cgroup, dest, grep_filters)
 
-    def tasks(self, cgroup):
+    def tasks(self, cgroup,
+              filter_tid=None,
+              filter_tname=None,
+              filter_tcmdline=None):
+        """
+        Report the tasks that are included in a cgroup. The tasks can be
+        filtered by their tid, tname or tcmdline if filter_tid, filter_tname or
+        filter_tcmdline are defined respectively. In this case, the reported
+        tasks are the ones in the cgroup that match these patterns.
+
+        Example of tasks format:
+        TID,tname,tcmdline
+        903,cameraserver,/system/bin/cameraserver
+
+        :params filter_tid: pattern to filter by TID
```

Comment:
Maybe we should specify that's a regexp?

Reply:
Should we also add a sanity check? It should be either None or a string...

Reply:
specify that is a regex: Done
i have added sanity checks at the start of the function for the filter. I change as well the default value of the filter for the empty string.